app/code/Magento/CatalogSearch/Model/Search/Catalog.php

```diff
@@ -33,58 +35,53 @@ class Catalog extends \Magento\Framework\DataObject
      *
      * @var \Magento\Backend\Helper\Data
      */
-    protected $_adminhtmlData = null;
+    protected $_adminHtmlData = null;
 
     /**
-     * @param \Magento\Backend\Helper\Data $adminhtmlData
+     * @param \Magento\Backend\Helper\Data $adminHtmlData
      * @param \Magento\Framework\Stdlib\StringUtils $string
      * @param QueryFactory $queryFactory
      */
     public function __construct(
-        \Magento\Backend\Helper\Data $adminhtmlData,
+        \Magento\Backend\Helper\Data $adminHtmlData,
         \Magento\Framework\Stdlib\StringUtils $string,
         QueryFactory $queryFactory
     ) {
-        $this->_adminhtmlData = $adminhtmlData;
+        $this->_adminHtmlData = $adminHtmlData;
         $this->string = $string;
         $this->queryFactory = $queryFactory;
     }
 
     /**
-     * Load search results
-     *
-     * @return $this
+     * {@inheritdoc}
      */
-    public function load()
+    public function getResults(SearchCriteria $searchCriteria)
```

Comment:
This change effectively removes method `load`, this violates backwards compatibility for the third-party code which may rely on it. 
You may add a new method and proxy calls from the old one, given that result will be compatible